tricycle/_tests/test_tree_var.py

```diff
@@ -136,8 +136,67 @@ async def run_sync(fn: Any, *args: Any) -> Any:
         tv1.get,
         partial(tv1.get, 20),
         partial(tv1.set, 30),
-        lambda: tv1.reset(trio.run(run_sync, tv1.set, 10)),
+        lambda: tv1.reset(tv1.set(10)),
         tv1.being(40).__enter__,
     ):
-        with pytest.raises(RuntimeError, match="must be called from async context"):
-            operation()  # type: ignore
+        operation()  # type: ignore
+    
+    diff_context_regex = r".*was created in a different Context"
+    # Resetting the context var in sync context to the token set in async context should raise
+    with pytest.raises(ValueError, match=diff_context_regex):
+        tv1.reset(trio.run(run_sync, tv1.set, 10))
+
+
+    tv2 = TreeVar("tv2", default=10)
+    # Assert that being also works in sync context
+    with tv2.being(30):
+        assert tv2.get() == 30
+    assert tv2.get() == 10
+
+    def mix_async_and_sync():
```

Comment:
This is my interpretation of its expected behavior when we mix sync and async context, let me know if any of it is not in line with what you have in mind!